internal/cmdrpkgclone/command.go

```diff
@@ -154,11 +145,15 @@ func (r *runner) preRunE(cmd *cobra.Command, args []string) error {
 		}
 
 	case strings.Contains(source, "/"):
+		gitArgs, err := parse.GitParseArgs(context.Background(), args)
```

Comment:
will this try to clone the repository locally (or try to contact it in any way), or is this purely syntax-level parsing?

Reply:
From what I can tell, `GitParseArgs` will try to contact the repository only if there is no ref/version provided in the URL. It tries to make contact in this case to determine what the default branch is. 

What is the correct behavior for `kpt alpha rpkg clone` if no version is provided? I can extract the syntax-level parsing out of `GitParseArgs` and use that here instead if we want to do something different.

Reply:
empty ref would be ok, or we can default to `main`. Porch defaults to `main` in other places where ref is not specified.

Error may be OK too.

I actually thought (though may be wrong) that another case where we contact the server is if the .git/ part is missing and the parser can't tell where the repository path ends and directory path begins:

domain.com/some/path/here

is the `some/path` repo? or `some` ?

regardless, the rpkg commands should not interact with the repository and only operate at the level of syntax of the URL provided, even if there are cases we can't handle and that will result in errors (IMO)

Reply:
> empty ref would be ok, or we can default to main. Porch defaults to main in other places where ref is not specified.

I'll default to `main` to be consistent with other places. 

> I actually thought (though may be wrong) that another case where we contact the server is if the .git/ part is missing and the parser can't tell where the repository path ends and directory path begins

Ah you are correct, I missed that it contacts the repo while attempting to convert the url to a URL containing `.git`.

Is it a requirement for `kpt alpha rpkg clone` that the URL contains `.git`? I'm thinking about moving the logic in `GitParseArgs` that handles URLs with a .git suffix out to its own helper function so that we can use it here without worrying about communication with the repo.

Reply:
Presence of the `.git` part of the URL is not requirement. I'd say the requirement is that the parser doesn't _modify_ the URL.

So if customer gave us https://github.com/org/repo.git/path/to/package
we should include the `.git`

if they provided https://github.com/org/repo/path/to/package
we should not artificially introduce `.git` part into the URL.